fs.go

```diff
@@ -297,7 +297,7 @@ func ParseTemplates(f template.FuncMap, fs FileSystem, path ...string) (*templat
 			return nil, fmt.Errorf("%s: %v", p, err)
 		}
 
-		_, err = tpl.Parse(string(f))
+		_, err = tpl.New(filepath.Base(p)).Parse(string(f))
```

Comment:
@knadh I had done some quick testing and it didn't seem to break anything. I'll be giving it a more thorough testing as soon as I work through some other tasks while upgrading listmonk.

The way I understand it to work, is that calling `tpl.New()` (as opposed to `template.New()`) creates a new template in the already existing `tpl` object. So within `tpl`, we are creating a template named `campaign-status.html`. However, that template is empty since we only `define "campaign-status"` in that file. I believe `html/template` ignores empty templates.

I think there's another advantage -- one could avoid the need to wrap every included file with a `{{ define }}` tag, and instead have templates based on the filename.

Reply:
Haven't had a chance to investigate this, but `go test` fails. Please try running it.

Reply:
Fixed failing tests and added new tests.